consensus/hotstuff/README.md

```diff
@@ -78,58 +78,62 @@ The figure below illustrates the dependencies of the core components and informa
-processing step via a direct method call (see `PaceMaker interface`). If the PaceMaker changed the view in response, it returns 
+* [synchronous] When progress is made following the core business logic, the  `EventHandler` will inform the PaceMaker about discovering new QCs or TCs 
+via a direct method call (see `PaceMaker interface`). If the PaceMaker changed the view in response, it returns 
 a `NewViewEvent` which will be synchronously processed by the  `EventHandler`.
 
-Flow's PaceMaker is simple (compared to Libra's PaceMaker for instance). It solely relies on increasing the timeouts 
-exponentially if no progress is made and exponentially decreasing timeouts on the happy path. 
+Flow's PaceMaker is an implementation of Active Pacemaker which communicates with other replicas after failing to make progress on happy path. It relies on broadcasting timeout object whenever no progress is made during current round. After collecting timeouts from supermajority of participants replica constructs a TC which can be used to enter next round `V = TC.View + 1`. For calculating round timeouts we use a truncated exponential backoff. We will increase round duration exponentially if no progress is made and exponentially decreasing timeouts on happy path. Protocol tolerates a few failures(`k=6`) before increasing round timeout. This value is the number of rounds we expect during happy path, after this many rounds, we will start increasing timeouts.
 The timeout values are limited by lower and upper-bounds to ensure that the PaceMaker can change from large to small timeouts in a reasonable number of views. 
-The specific values for lower and upper timeout bounds are protocol-specified; we envision the bounds to be on the order of 1sec (lower bound) and multiple days (upper bound).
+The specific values for lower and upper timeout bounds are protocol-specified; we envision the bounds to be on the order of 1sec (lower bound) and one minute (upper bound).
 
 **Progress**, from the perspective of the PaceMaker is defined as entering view `V`
-for which the replica knows a QC with `V = QC.view + 1`. 
+for which the replica knows a QC or a TC with `V = QC.view + 1` or `V = TC.view + 1`. 
 In other words, we transition into the next view due to reaching quorum in the last view.
+When comparing to HotStuff, in Jolteon views can be changed only after observing a valid quorum in the last view, otherwise replica is not allowed to change view. 
   
 A central, non-trivial functionality of the PaceMaker is to _skip views_. 
-Specifically, given a QC with view `qc.view`, the Pacemaker will skip ahead to view `qc.view + 1` if `currentView ≤ qc.view`.
-  
-<img src="https://github.com/onflow/flow-go/blob/master/docs/PaceMaker.png" width="200">
+Specifically, given a QC or TC with view `V` or , the Pacemaker will skip ahead to view `V + 1` if `currentView ≤ V`.
+
+![](/docs/PaceMaker.png)
```

Comment:
In the first yes branch for the TC flow diagram, text should be `curView = tc.View + 1`